lsp-mode.el

```diff
@@ -7081,22 +7109,24 @@ When prefix UPDATE? is t force installation even if the server is present."
 
 (defun lsp-async-start-process (callback error-callback &rest command)
   "Start async process COMMAND with CALLBACK and ERROR-CALLBACK."
-  (make-process
-   :name (cl-first command)
-   :command command
-   :sentinel (lambda (proc _)
-               (when (eq 'exit (process-status proc))
-                 (if (zerop (process-exit-status proc))
-                     (condition-case err
-                         (funcall callback)
-                       (error
-                        (funcall error-callback (error-message-string err))))
-                   (display-buffer " *lsp-install*")
-                   (funcall error-callback
-                            (format "Async process '%s' failed with exit code %d"
-                                    (process-name proc) (process-exit-status proc))))))
-   :buffer " *lsp-install*"
-   :noquery t))
+  (let ((name (cl-first command)))
+    (with-current-buffer (compilation-start (combine-and-quote-strings command) t
+                                            (lambda (&rest _)
+                                              (generate-new-buffer-name (format "*lsp-install: %s*" name))))
+      (setq-local lsp--installation-buffer t)
+      (add-hook
+       'compilation-finish-functions
+       (lambda (buf status)
+         (if (string= "finished\n" status)
+             (condition-case err
+                 (funcall callback)
+               (error
+                (funcall error-callback (error-message-string err))))
+           (funcall error-callback
+                    (format "Async process '%s' failed with exit code %d"
+                            name (-> (current-buffer) get-buffer-process
```

Comment:
Shouldn't this be `buf`?

Reply:
This was eliminated, because after compilation finishes, `get-buffer-process` always returns nil. I decided to simply use the `status` string as the argument to error-callback.